spec/requests/admin/justifications_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Admin justifications', type: :request do
+  before do
+    sign_in admin
+  end
+
+  let(:admin) { create(:admin) }
+  let(:student) { create(:student) }
+  let(:justifications) { create_list(:justification, 3, email: student.email) }
+
+  describe '#index' do
+    before do
+      get admin_justifications_path
+    end
+
+    it 'renders the index template' do
+      expect(response).to render_template(:index)
+    end
+
+    it 'shows all the student justifications' do
+      expect(justifications.count).to eq(3)
+    end
+  end
+
+  describe '#show' do
+    it 'shows a justification' do
+      get admin_justification_path(justifications.first.id)
+      expect(response).to be_successful
+    end
+  end
+
+  describe '#update' do
+    it 'updates the justification status' do
+      put admin_justification_path(justifications.first.id), params: { justification: { status: 'aceito' } }
+      expect(justifications.first.status).to eq('aceito')
+      expect(response).to redirect_to admin_justifications_path
+    end
+  end
+
+  describe '#destroy' do
+    it 'deletes a justification' do
+      delete admin_justification_path(justifications.first.id)
+      expect(response).to redirect_to admin_justifications_path
+    end
```

Comment:
@davithss this spec is generating a false positive. For example, if something went wrong during the initialization or delete action on the controller, the redirecting action will still be happening so your tests you be passing but your controller could not be deleting the record.

You should improve this to two scenarios:
1) You expect that the number of justifications decreases by one
2) Redirect to `admin_justifications_path`